### PR TITLE
Adding auth method to auth svc

### PR DESCRIPTION
**Pull request by georific**
## Status
**READY**

## JIRA Ticket
Adding ability to interchange different auth methods more easily in the auth service.

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] License

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

```sh
git fetch --all
git checkout <feature_branch> 
./gradlew connectedCheck
```

@boundlessgeo/spatial-connect

### CODE REVIEW

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/SpatialConnect.java [hunk 1]
```diff
@@ -228,6 +228,12 @@ public void connectBackend(final SCRemoteConfig remoteConfig) {
         }
     }
 
+    public void connectAuth(ISCAuth authMethod) {
+        authService = new SCAuthService(context, authMethod);
+        addService(authService);
+        startService((SCAuthService.serviceId()));
```
Comment (mrcnc) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/SpatialConnect.java]: Looks like you have some extra parentheses here?  Must be all the LISP you've been reading 😉

Comment (georific) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/SpatialConnect.java]: haha good CATCH!

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/ISCAuth.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.boundlessgeo.spatialconnect.services.authService;
```
Comment (mrcnc) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/ISCAuth.java]: Can you put the license at the top and then the package after it?  I believe that's been our convention with the other files

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCServerAuthMethod.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import okhttp3.Response;
+
+/**
```
Comment (mrcnc) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCServerAuthMethod.java]: Here too...can you add it to the top of the file?

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCServerAuthMethod.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        editor.putString(PWD, password);
+        editor.commit();
+    }
+
+    private String getPassword() {
+        SecureSharedPreferences settings = new SecureSharedPreferences(context);
+        return settings.getString(PWD, null);
+    }
+
+    private String getUsername() {
```
Comment (mrcnc) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCServerAuthMethod.java]: Maybe we should get rid of this method and just use `username()` b/c that's what's required by the interface?

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCServerAuthMethod.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+                    saveCredentials(username, pwd);
+                    authed = true;
+                }
+            } catch (JSONException e) {
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            authed = false;
```
Comment (mrcnc) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCServerAuthMethod.java]: Won't `authed` always be false anytime it gets to this line?  If so, we can probably remove it b/c it's redundant

Comment (georific) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCServerAuthMethod.java]: indeed

**Comment by mrcnc** [comment]
👍

**Pull request merged**